astropy/io/ascii/ui.py

```diff
@@ -627,6 +627,37 @@ def _guess(table, read_kwargs, format, fast_reader):
         cparser.CParserError,
     )
 
+    # Determine whether we should limit the number of lines used in the guessing.
+    # Note that this does not currently work for file objects, so we set this to
+    # False if a file object was passed in.
+    from astropy.io.ascii import conf  # avoid circular imports
+
+    limit_lines = conf.guess_limit_lines and not hasattr(table, 'read')
```

Comment:
```suggestion
    limit_lines = conf.guess_limit_lines * not hasattr(table, 'read')
```
or
```suggestion
    limit_lines = conf.guess_limit_lines if not hasattr(table, 'read') else False
```
This was setting `limit_lines = True`, which caused only the first line without newline to be read into `table_guess_subset`.